### DISCUSSION TITLE
Modal problem

### DISCUSSION THREAD

**Discussion by gotzsys** [filamentphp/filament#6515] [Help]
Hello everyone,

I am witnessing a performance issue when opening a modal within a table. The size of the loaded file increases as the number of records listed in the grid increases.
![image](https://github.com/filamentphp/filament/assets/13874850/ddb40bf1-8fc9-4880-8bdb-f2701d524bc4)

As I keep adding more records, this size keeps increasing without stopping.
![image](https://github.com/filamentphp/filament/assets/13874850/ee0ff379-26ca-40ad-a1cb-2264e806d782)

With 50 records being loaded, this size keeps increasing without stopping.
![image](https://github.com/filamentphp/filament/assets/13874850/6f807cb9-8895-4430-8c39-85cdc81ca2e8)

```
IconColumn::make('mobile')
    ->label('')
    ->trueIcon('heroicon-o-chat')
    ->color('success')
    ->action(
        Action::make('sendMessage')
            ->label('Enviar mensagem')
            ->form([
                Toggle::make('is_vip')
                    ->label('Clube VIP')
                    ->inline(true)
                    ->onColor('success')
                    ->offColor('second')
                    ->columnSpanFull(),
            ])
    ),
```

**Comment by danharrin** [comment]
yeah, this is just how Livewire works. the size you're seeing is how much HTML there is. as the table grows, more HTML is required to render it.

**Comment by gotzsys** [comment]
Hello Dan,
These requests were not made during the table's initial loading, but rather when opening the modal itself. When I open the modal, does it reload all the table information in parallel?

**Reply by danharrin** [reply]
yes it does, its all part of the same LW component

**Comment by gotzsys** [comment]
I understand, Dan. I will try to find a way to work around this issue.
Tks

**Reply by underdpt** [reply]
Hello,

I'm having this exact issue. It's a bit of a problem when working with huge tables, as clicking on an action takes ages, and it gets worse if the user is showing lots of rows.

Did you find a way to solve this?

I'm thinking on getting back to wire-elements/modal (as I had on Filament 2) to check if that would solve this issue.
